### DISCUSSION TITLE
Should we start mandating that Pull Requests update the CHANGELOG?

### DISCUSSION THREAD

**Discussion by lovelydinosaur** [encode/httpx#1504] [General]
So, I've been thinking that we might have reached a point in our workflow where we'd like pull requests to also include a relevant CHANGELOG entry.

We're pretty low churn at this point, plus GitHub's UI now makes it super easy to resolve the sort of simple merge conflicts that CHANGELOG-entries-in-PRs often introduce.

If we forced pull requests to include CHANGELOG entries, then...

* We're enforcing that all behaviour-changing pull requests get a nicely worded single-line motivation description.
* Our release process & reviewing potential breaking changes etc. becomes easier.
* History-wise the CHANGELOG entries are nicely associated alongside with the same PR that introduced them.

(Note that refactors and docs changes *don't* require CHANGELOG entries, since they're not behavioural changes.)

**Comment by florimondmanca** [comment]
Personally don't mind experimenting with that for a bit. :-)

Right now maintainers have to go through the git history and decide which PRs should go into a release and which shouldn't. I don't think the cumbersomeness of it is a problem, but it certainly is a bit risky — might miss important PRs, not see what's breaking and what's not, etc.

How would we _enforce_ this new policy?

* Have maintainers "remember" about it? (Seems unsustainable?)
* Add a PR template with a "This PR contains any appropriate changelog entry" checkbox? Related https://github.com/encode/httpx/discussions/1493#discussioncomment-420881

How do we feel about something like [Reno](https://pypi.org/project/reno/)?

**Reply by lovelydinosaur** [reply]
Re: Reno - As always I'm a super-nope on extra tooling, so...

A pull request template makes a lot of sense yup. We could do with one anyways. (And start pushing folks towards discussions as a typical pre-cursor to issues/PRs)

**Reply by florimondmanca** [reply]
> super-nope on extra tooling

😆

Yes, a PR template has become definitely needed for HTTPX (and HTTPCore?).

**Reply by StephenBrown2** [reply]
Reno might be good, but I think something like [towncrier](https://github.com/twisted/towncrier) would be simpler, as it only requires PRs to include a single file in a directory describing the change. Any changelog tooling would only need to live in GH Actions in my opinion. However, if you're against more dependency tooling, we could copy what [Pydantic has done](https://github.com/samuelcolvin/pydantic/tree/master/changes) with a [single "make_history" file](https://github.com/samuelcolvin/pydantic/blob/master/changes/make_history.py) and again, just a single file for contributers to add.

PR Template++ regardless though.
